s3/src/test/io/ddf/s3/S3DDFManagerTests.java

```diff
@@ -0,0 +1,83 @@
+package io.ddf.s3;
+
+import io.ddf.DDFManager;
+import io.ddf.datasource.DataFormat;
+import io.ddf.datasource.S3DataSourceCredentials;
+import io.ddf.datasource.S3DataSourceDescriptor;
+import io.ddf.datasource.S3DataSourceURI;
+import io.ddf.exception.DDFException;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+
+public class S3DDFManagerTests {
+
+    public static S3DDFManager manager;
+    public static Logger LOG;
+
+    @Test
+    public void testDDFConfig() throws Exception {
+        Assert.assertEquals("s3", manager.getEngine());
+    }
+
+    @BeforeClass
+    public static void startServer() throws Exception {
+        Thread.sleep(1000);
+        LOG = LoggerFactory.getLogger(S3DDFManagerTests.class);
+        S3DataSourceDescriptor  s3dsd = new S3DataSourceDescriptor(new S3DataSourceURI(""),
+            new S3DataSourceCredentials(System.getenv("AWS_ACCESS_KEY_ID"), System.getenv("AWS_SECRET_ACCESS_KEY")),
+            null,
+            null);
+        manager = (S3DDFManager)DDFManager.get(DDFManager.EngineType.S3, s3dsd);
+    }
+
+    @Test
+    public void testListing() throws DDFException {
```

Comment:
Yeah I know that, I mean a test that asserts list return something like:
subfolder/
file1.csv
file2.csv